sls_api/endpoints/generics.py

```diff
@@ -809,3 +810,95 @@ def handle_deleted_flag(values: Dict[str, Any]) -> Dict[str, Any]:
     if values.get("deleted"):
         values["published"] = 0
     return values
+
+
+def is_valid_year(year_string: str) -> bool:
+    """
+    Checks if a string can be parsed as a four-digit year between 1 and 9999.
+
+    The function validates that the input string consists of only digits and
+    represents a year between 1 and 9999. It handles both zero-padded and
+    non-zero-padded formats for years less than 1000 (e.g., "0456" and "456"
+    are both valid).
+
+    Args:
+
+        year_string (str): The input string representing a year.
+
+    Returns:
+
+        bool: True if the string represents a valid year between 1 and 9999,
+        False otherwise.
+    """
+    if not year_string.isdigit():
+        return False
+
+    # Check if the integer value is between 1 and 9999
+    year = int(year_string)
+    return 1 <= year <= 9999
+
+
+def is_valid_date(date_string: str) -> bool:
+    """
+    Validates if a given string conforms to the 'YYYY-MM-DD' date format
+    and checks if it represents a logically date.
+
+    Parameters:
+
+        date_string (str): The input string to be checked.
+
+    Returns:
+
+        bool: True if the string is a valid 'YYYY-MM-DD' date format and
+        represents a logically correct date; False otherwise.
+
+    Examples:
+
+        >>> is_valid_date("2023-10-31")
+        True
+        >>> is_valid_date("2023-02-29")
+        False  # 2023 is not a leap year
+        >>> is_valid_date("2023-13-31")
+        False  # Invalid month
+        >>> is_valid_date("23-10-31")
+        False  # Invalid format
+    """
+    try:
+        datetime.strptime(date_string, "%Y-%m-%d")
+        return True
+    except ValueError:
+        return False
+
+
+def is_valid_year_month(date_string: str) -> bool:
+    """
+    Validates if a given string conforms to the 'YYYY-MM' date format.
+    If the YYYY part is a year before 1000, it must be zero-padded.
+
+    Depends on the `is_valid_year()` helper function.
+
+    Parameters:
+
+        date_string (str): The input string to be checked.
+
+    Returns:
+
+        bool: True if the string is a valid 'YYYY-MM' format; False
+        otherwise.
+
+    Examples:
+
+        >>> is_valid_year_month("2023-10")
+        True
+        >>> is_valid_year_month("2023-13")
+        False  # Invalid month
+    """
+    parts = date_string.split("-")
```

Comment:
Just use datetime.strptime here as well, rather than splitting and doing manual integer parsing.